src/App.tsx

```diff
@@ -15,48 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+function getOrderedGoods(
+  goods: string[],
+  orderedMethod: SortType,
+  isReversed: boolean,
+) {
+  const finalGoods = [...goods];
+
+  switch (orderedMethod) {
+    case SortType.ALPHABET:
+      finalGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SortType.LENGTH:
+      finalGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    finalGoods.reverse();
+  }
+
+  return finalGoods;
+}
+
 export const App: React.FC = () => {
+  const [orderMethod, setOrderMethod] = useState(SortType.DEFAULT);
```

Comment:
```suggestion
  const [order, setOrder] = useState(SortType.DEFAULT);
```
why method? or `sortType, setSortType`, etc